### DISCUSSION TITLE
Submitted filter seems broken

### DISCUSSION THREAD

**Discussion by kc8se** [community/community#143919] [GitHub Education]
### Hi everyone,

First of all, thanks for GitHub Education, it's really helpful and allows the students to interact with GitHub (often their first time) in a meaningful way.

All that said, I thought I'd help out and report some bugs/improvements:

1. "Submitted" and "Not Submitted" seem broken (and they never worked in my experience). The description of "Submitted" is "Students who have commited (and pushed / my note) to the repository". That is a very useful feature. But it has false positives. Here is a link to a pastebin with links to an assignment and student who is a false positive: https://pastebin.com/raw/F7Eu66SS
2. If you have selected to filter by Passing or Failing, then you have to click Clear Current Search Query to get out of that filter. I think it would be good UX to have "Both" or similar as an option too. When an assignment does not have autograding tests then both Passing and Failing shows none.

And I also have a small Wish List: :)

1. "Unread" would be nice to have as a filter, and perhaps shown on each assignment submission line. It would highlight if the student have written a comment after my last comment. Or if I haven't written a comment, it would just show if the student has. It would make it easier to see who wants to communicate even when they haven't tagged me.
2. "Changes Requested" to indicate if I my latest review of the Feedback pull request was a request to the student for changes.
3. "Retry" to indicate if there has been pushes since I set the status as described above to Changes Request.
4. "Review needed" would show all Submitted which does not have Accepted as my last review of the Feedback PR. It would filter out those who have status "Changes Requested" and not "Retry".

The wish list would make me much more effective. I have plans to write a small CLI tool for the above, but I wouldn't have to any everyone can enjoy it if it would be built in.

Thanks again! As this is a discussion, let's discuss the above as well. :)

**Comment by MisterBrash** [comment]
Seeing as you posted this back in November and nobody has replied, I don't think the good people at GitHub Classroom are able to keep up with the sheer number of posts in these groups.

Your issue is similar to: https://github.com/orgs/community/discussions/137066

I also have noticed that the "submitted" tag is broken. Totally broken.

If an assignment has a due date, the bot creates at least one commit on the repo after the student accepts it - flagging the assignment as submitted. This means that my entire class is listed as submitted after having _just started_ the assignment.

I can not use that flag as anything other than a nice green colour in the list of students - it is meaningless.

I wish there was a real way for users of GH Classroom to contact the devs or management team at GitHub to give real feedback other than these discussion groups that don't get looked-at...

**Reply by kc8se** [reply]
Indeed. It doesn't seem to be a prioritized project for GitHub.

My students closed the "Feedback" PR more often than I liked so I have a cron job which protects it now. It's kind of hacky, but gets the job done. It would be nice to have such things included from the start. Still, beggars can't be choosers I guess.
